### ISSUE TITLE
Knobs stopped handling object values properly

### ISSUE THREAD

**Issue by Hypnosphi** [storybooks/storybook#4500]
```
import React from 'react'
import {storiesOf} from '@storybook/react'
import {withKnobs, select} from '@storybook/addon-knobs'

const options = {
  first: {foo: 'foo},
  second: {bar: 'bar'},
}

storiesOf('MyComponent', module)
  .addDecorator(withKnobs)
  .add('MyComponent', () => (
    <MyComponent
      object={select('Option', options, options.second)}
    />
  )
```

Expected: "second" selected in knob
Actual: "first" selected in knob
After trying to select other value in knob, `"[object Object]"` gets passed to `MyComponent`

The workaround is to use object keys instead:
```
<MyComponent
  object={options[select('Option', Object.keys(options), 'second')]}
/>
```

**Comment by lapa182** [comment]
Is there a feedback with this issue? Having the same problem.

**Comment by Hypnosphi** [comment]
I believe it was fixed with #4522. Which storybook version do you use?
